classes/UpgradeContainer.php

```diff
@@ -289,8 +289,10 @@ public function getUpgrader()
             define('_PS_ROOT_DIR_', $this->getProperty(self::PS_ROOT_PATH));
         }
         // in order to not use Tools class
-        $upgrader = new Upgrader($this->getProperty(self::PS_VERSION));
-        preg_match('#([0-9]+\.[0-9]+)(?:\.[0-9]+){1,2}#', $this->getProperty(self::PS_VERSION), $matches);
+        $currentVersion = $this->getProperty(self::PS_VERSION);
+        $upgrader = new Upgrader($currentVersion);
+        $this->getState()->setOriginVersion($currentVersion);
+        preg_match('#([0-9]+\.[0-9]+)(?:\.[0-9]+){1,2}#', $currentVersion, $matches);
```

Comment:
We maybe must check if $matches is not empty. something broken can came from the api :/

Reply:
In this case since the version comes from `UpgraderContainer::getProperty` which uses either constants or files content, so we should be safe.
`PrestashopConfiguration::getPrestaShopVersion` already throws an exception if it can't find the version